src/Microsoft.Azure.WebJobs.Host/Listeners/HostListenerFactory.cs

```diff
@@ -70,7 +74,19 @@ public async Task<IListener> CreateAsync(CancellationToken cancellationToken)
 
                 IListener listener = await listenerFactory.CreateAsync(cancellationToken);
 
-                RegisterScaleMonitor(listener, _monitorManager);
+                bool targetScalerRegistered = false;
+                if (bool.TryParse(Environment.GetEnvironmentVariable(TargetBaseScalingEnabled), out bool isTargetBaseScalingEnabled))
```

Comment:
In addition to the `TARGET_BASED_SCALING_ENABLED` app setting, we also need to check if TBSA is enabled (per extension type) via the `ScaleControllerFeatureFlags` hosting configuration. See [Scale Controller PR](https://msazure.visualstudio.com/One/_git/AAPT-Antares-ScaleController/pullrequest/6552084) and [TBS +RDS design doc](https://microsoft-my.sharepoint.com/:w:/p/vchiang/Edf-XUIR7ZpHuwhZYporzlUB3UkSnjPMV9Wz94UPU4bV4g?e=9FwGKq) for reference

Reply:
I wonder if we shouldn't just remove this env var check from the SDK. SDK should just register all monitors and scalers, and it's up to the consumer (Functions Host) to determine whether based on environment they should be used. Note that's how monitors already work - SDK registers them, but Functions Host only uses them based on FUNCTIONS_RUNTIME_SCALE_MONITORING_ENABLED

Reply:
Moved both checks (TARGET_BASED_SCALING_ENABLED  and hosting configuration per individual trigger) to function host